File: Source/ContactManager/Models/Domain/UserPropertiesModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models.Domain
    ''' <summary>
    ''' Expose public properties for all user properties created for this application.
    ''' </summary>
    Public Class UserPropertiesModel
        '
        Private cp As CPBaseClass
        '
        '
        '====================================================================================================
        ''' <summary>
        ''' constructor
        ''' </summary>
        ''' <param name="cp"></param>
        Public Sub New(cp As CPBaseClass)
            Me.cp = cp
        End Sub
        '
        '====================================================================================================
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <returns></returns>
        Public Property contactSearchCriteria As String
            Get
                Return cp.TempFiles.Read("contactManager\user-contact-search-criteria.txt")
            End Get
            Set(value As String)
                cp.TempFiles.Save("contactManager\user-contact-search-criteria.txt", value)
            End Set
        End Property
        '
        '====================================================================================================
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <returns></returns>
        Public Property contactGroupCriteria As String
            Get
                Return cp.TempFiles.Read("contactManager\user-group-search-criteria.txt")
            End Get
            Set(value As String)
                cp.TempFiles.Save("contactManager\user-group-search-criteria.txt", value)
            End Set
        End Property
        '
        '====================================================================================================
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <returns></returns>
        Public Property selectSubTab As Integer
            Get
                Return cp.User.GetInteger(nameSelectSubTab)
            End Get
            Set(value As Integer)
                cp.User.SetProperty(nameSelectSubTab, value)
            End Set
        End Property
        Private Const nameSelectSubTab As String = "SelectSubTab"
        '
        '====================================================================================================
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <returns></returns>
        Public Property subTab As Integer
            Get
                Return cp.User.GetInteger(nameSubTab, 1)
            End Get
            Set(value As Integer)
                cp.User.SetProperty(nameSubTab, value)
            End Set
        End Property
        Private Const nameSubTab As String = "SubTab"
        '
        '====================================================================================================
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <returns></returns>
        Public Property ContactContentID As Integer
            Get
                Return cp.User.GetInteger(nameContactContentID, cp.Content.GetID("people"))
            End Get
            Set(value As Integer)
                cp.User.SetProperty(nameContactContentID, value)
            End Set
        End Property
        Private Const nameContactContentID As String = "ContactContentID"
        '
        '
        '
    End Class
End Namespace

File: Source/ContactManager/Models/Domain/UserPropertiesModel.cs
using Contensive.BaseClasses;

namespace Contensive.Addons.ContactManager.Models.Domain {
    /// <summary>
    /// Expose public properties for all user properties created for this application.
    /// </summary>
    public class UserPropertiesModel {
        // 
        private readonly CPBaseClass cp;
        // 
        // 
        // ====================================================================================================
        /// <summary>
        /// constructor
        /// </summary>
        /// <param name="cp"></param>
        public UserPropertiesModel(CPBaseClass cp) {
            this.cp = cp;
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public string contactSearchCriteria {
            get {
                return cp.TempFiles.Read(@"contactManager\user-contact-search-criteria.txt");
            }
            set {
                cp.TempFiles.Save(@"contactManager\user-contact-search-criteria.txt", value);
            }
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public string contactGroupCriteria {
            get {
                return cp.TempFiles.Read(@"contactManager\user-group-search-criteria.txt");
            }
            set {
                cp.TempFiles.Save(@"contactManager\user-group-search-criteria.txt", value);
            }
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public int selectSubTab {
            get {
                return cp.User.GetInteger(nameSelectSubTab);
            }
            set {
                cp.User.SetProperty(nameSelectSubTab, value);
            }
        }
        private const string nameSelectSubTab = "SelectSubTab";
        // 
        // ====================================================================================================
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public int subTab {
            get {
                return cp.User.GetInteger(nameSubTab, 1);
            }
            set {
                cp.User.SetProperty(nameSubTab, value);
            }
        }
        private const string nameSubTab = "SubTab";
        // 
        // ====================================================================================================
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public int contactContentID {
            get {
                return cp.User.GetInteger(nameContactContentID, cp.Content.GetID("people"));
            }
            set {
                cp.User.SetProperty(nameContactContentID, value);
            }
        }
        private const string nameContactContentID = "ContactContentID";
        // 
        // 
        // 
    }
}